### PR TITLE
Bug 1798382 - Bugzilla queries that involve change history are rejected by MySQL when using date shortcuts and  the calculated timestamp is an invalid date

### PR DESCRIPTION
**Pull request by dklawren**
Removes all manual date math and instead uses DateTime for date math.

### CODE REVIEW

File: Bugzilla/Search.pm [hunk 1]
```diff
@@ -2318,10 +2319,10 @@ sub _date_translate {
 sub SqlifyDate {
   my ($str) = @_;
   my $fmt = "%Y-%m-%d %H:%M:%S";
+  my $date = DateTime->now();
```
Comment (globau) [inline Bugzilla/Search.pm:2322]: Should we be using the database's current time rather than the webheads?

Comment (dklawren) [reply Bugzilla/Search.pm:2322]: All pods should be in sync time wise but there could be some difference. I thought about it and was trying to keep the extra SQL calls to a minimum to get the current timestamp each time SqlifyDate is called. But maybe in would be insignificant? Thoughts?

Comment (dklawren) [reply Bugzilla/Search.pm:2322]: I take that back, the MySQL server itself is a separate instance and may not be time synced.

Comment (globau) [reply Bugzilla/Search.pm:2322]: Fair enough, let's leave it client-side.

**Review by globau — APPROVED** [review]

**Pull request merged**